SparkPost/2web2ui
# FE-765 - Adds React Testing Library

[FE-765](https://jira.int.messagesystems.com/browse/FE-765)

### What Changed
- Installs `@testing-library/react` ➡️ https://github.com/testing-library/react-testing-library
- Installs supporting matcher library `@testing-library/jest-dom` ➡️ https://github.com/testing-library/jest-dom
- Installs supporting user event simulation library `@testing-library/user-event` for typing and click simulation ➡️ https://github.com/testing-library/user-event
- Installs supporting ESLint plugin to encourage best practices `eslint-plugin-testing-library` ➡️ https://github.com/Belco90/eslint-plugin-testing-library
- Added `data-id` attribute to `<PanelLoading/>` component
- Configured default test ID attribute to `data-id` instead of `data-testid`
- Updated Jest setup script to catch warnings with exception for bug with React 16.8.0 - see comments for more information
- Filed new issue with Matchbox regarding use of `data-test` instead of `data-id` in that library
- Updated `<SendTestEmailButton/>` component tests using React testing library both as an example and to prove that the new setup functions as expected

### How To Test
- Checkout this branch, run `npm install`
- Run all tests and verify that they continue to pass
- Write a new test anywhere in the app using React Testing Library

### To Do
- [ ] Update Wiki documentation to discourage use of Enzyme...?
